docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,126 @@
+- LinkerD has only tutorials on how to [export metrics](https://linkerd.io/2.14/tasks/exporting-metrics/#proxy), and creating a service profile can enable per route metrics
+
+#### New API suggestion:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetrics
+metadata:
+  name: default
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset | MeshService | MeshServiceSubset
+  default:
+    filter:
+      profile: minimal
+      regex: http2_act.* # only profile or regex can be defined
+      usedOnly: true
+    aggregate:
+      - path: "/metrics/prometheus" # is name still needed since targetRef can just point to a name?
+        port: 8888
+      - port: 8000 # default path is /metrics
+    backends:
+      - type: Prometheus
+        prometheus:
+          port: 5670
+          path: /metrics
+          tls:
+            mode: disabled | providedTLS | activeMTLSBackend
+      - type: OpenTelemetry
+        openTelemetry:
+          endpoint: otel-collector:4317
+```
+
+### Configure MADS based on policy data
+MADS server is responsible for aggregating information about dataplanes needed for Prometheus monitoring. 
+It now extracts information from the `Mesh` resource which proxy should be monitored and creates a `MonitoringAssignment` response.
+
+We need to change the source of our information about monitored proxies from `Mesh` to `MeshMetric` policy. We cannot treat 
+new policy as usual in this context, we should not merge policies, but create different `MonitoringAssignment` for each 
+defined policy, this is needed for scenarios like migrating to a new backend. For example, when we define two `MeshMetric` 
+policies that target the same data plane proxy, we should create two separate `MonitoringAssignment` responses. 
+We can achieve this by extending Prometheus integration, so that it sends backend name with requests to MADS API.
+
+### Pushing metrics from DP to OTEL collector
+In order to push metrics to the OTEL collector, and preserve current Prometheus metrics, we need to start using [OTEL metrics sdk](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric). 
+It allows us to export metrics in multiple formats, especially the ones that we are interested in, Prometheus and OTLP. 
+Thanks to this, we will be able to preserve our old functionality and add the possibility to push metrics to the OTEL collector.
+
+### Metrics hijacker
+To preserve metrics hijacker functionality, we need to implement our own OTEL [Producer](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#Producer). 
+It needs to make HTTP requests to Envoy to gather stats, apply our custom transformations and convert them to OTLP format. 
+Thanks to usage of [PeriodicReader](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#PeriodicReader) 
+from OTEL SDK, we will be able to make these calls periodically. This reader can use both Prometheus and OTLP exporters 
+to publish metrics to desired backends.
```

Comment:
I think we need a schema to show everything here. What exposes what and how :)

Reply:
I've added diagram to ilustrate it